### PR TITLE
Refactor `SettingsImporter` [1/x]

### PR DESCRIPTION
**Pull request by cketti**
A first step in breaking `SettingsImporter` apart. I have several more refactoring steps planned.

This is motivated by issue #7682 which requires changes to the way we import server settings. While we could modify the current code to support this, I believe the change will be much easier once `SettingsImporter` has been refactored (and converted to Kotlin).

### CODE REVIEW

File: app/core/src/main/java/com/fsck/k9/preferences/SettingsFileParser.kt [hunk 1]
```diff
@@ -1,414 +1,441 @@
-package com.fsck.k9.preferences;
```
Comment (wmontwe) [inline app/core/src/main/java/com/fsck/k9/preferences/SettingsFileParser.kt]: I think we should split the Java to Kotlin conversion from the rafactoring into separate commits. With both mixed it is really hard to follow up on the changes.

**Review by wmontwe — APPROVED** [review]
I think it's good to have the parsing logic extracted. But it would be good to split the Java to Kotlin conversion from refactorings for easier reviews. Especially as the changed code is quite a lot and it's hard to decide if there are intended behaviour changes or not.

I leave it as is to not further delay this and assume the test is sufficiently covering the changes.

File: app/core/src/main/java/com/fsck/k9/preferences/SettingsFileParser.kt [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+                        }
+                    }
+                    else -> {
+                        skipElement()
+                    }
+                }
+            }
+        }
+
+        return accounts.takeIf { it.isNotEmpty() }
```
Comment (wmontwe) [inline app/core/src/main/java/com/fsck/k9/preferences/SettingsFileParser.kt:195]: Could be
```suggestion
        return if (accounts.isNotEmpty()) accounts else null
```

**Comment by cketti** [comment]
I've now split the conversion to Kotlin and the refactoring. However, the refactoring is pretty extensive. So the diff is still not pleasant to review.

File: app/core/src/main/java/com/fsck/k9/preferences/SettingsFileParser.kt (hunk 2 above)
Comment (cketti) [reply app/core/src/main/java/com/fsck/k9/preferences/SettingsFileParser.kt:195]: I think once one got used to `takeIf`, that variant is easier to read and write.
